how_to_contribute.qmd

```diff
@@ -46,17 +48,33 @@ We strongly encourage those who have created or are developing code to calculate
 
 Analysis workflows can be adapted to BON in a Box pipelines using a few simple steps. Once BON in a Box is installed on your computer, Pipelines can be edited and run locally by creating files in your local BON-in-a-box folder that was cloned from the GItHub repository. Workflows can be in a variety of languages, including R, Julia, and Python, and not all steps of the pipeline need to be in the same language.
 
-Analysis workflows can be converted to BON in a Box pipelines with a few steps:
+## Before you start
+
+Make sure that you are working in the right environment when you are running code in BON in a Box. This can be in your code editor of choice, such as visual studio code opened to the BON in a Box pipelines folder that has been cloned to your computer from github or an R studio project that is set up with the BON in a Box github repository. This will make it easier to pull and push to the github repository later.
+
+## Step 1: Plan your pipeline
+
+The first step is to sketch out what your pipeline will look like. This includes breaking up your workflow into logical steps that perform one task and could potentially be reused in other pipelines. Think of the inputs and outputs that you want each script to have and how you want to connect these into a full pipeline.
+
+Here is an example:
+
+![](images/Screenshot 2025-02-26 161306.png)
 
-## Step 1: Creating a YAML file
+Check if any of the steps you need are already available in BON in a Box to avoid having to rewrite them.
+
+## Step 2: Break up your scripts
+
+You can start writing the individual scripts in your pipeline. If you already have an analysis workflow, break this up into individual scripts representing the steps of the pipeline. Plan what you want the inputs and outputs to be for each script.
+
+## Step 3: Create a YAML file
 
 Each script should be accompanied by a YAML file (.yml extension) that specifies inputs and outputs of that step of the pipeline.
 
-Markdown is supported for pipeline, script, input and output descriptions. See this [CommonMark quick reference](https://commonmark.org/help/) for allowed markups.
+Markdown is supported for pipeline, script, input and output descriptions. See this [CommonMark quick reference](https://commonmark.org/help/) for allowed markups. These can include special characters, links, and photos.
 
 ### Name and description
 
-YAML files should begin with a name, description, and authors of the script. The name should be the same as the script that it describes but with a .yml extension.
+YAML files should begin with a name, description, and authors of the script. The name should be the name of the script described by the YAML file. Save the file with the same name as the script that it describes but with a .yml extension.
```

Comment:
Not sure what you mean by "The name should be the name of the script described by the YAML file."